aiokafka/partitioner.py

```diff
@@ -0,0 +1,96 @@
+import random
+
+
+class DefaultPartitioner(object):
+    """Default partitioner.
+
+    Hashes key to partition using murmur2 hashing (from java client)
+    If key is None, selects partition randomly from available,
+    or from all partitions if none are currently available
+    """
+
+    @classmethod
+    def __call__(cls, key, all_partitions, available):
+        """
+        Get the partition corresponding to key
+        :param key: partitioning key
+        :param all_partitions: list of all partitions sorted by partition ID
+        :param available: list of available partitions in no particular order
+        :return: one of the values from all_partitions or available
+        """
+        if key is None:
+            if available:
+                return random.choice(available)
+            return random.choice(all_partitions)
+
+        idx = murmur2(key)
+        idx &= 0x7FFFFFFF
+        idx %= len(all_partitions)
+        return all_partitions[idx]
+
+
+# https://github.com/apache/kafka/blob/0.8.2/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L244
+def murmur2(data):
```

Comment:
Is the reason for using this and not [mmh3](https://github.com/hajimes/mmh3#mmh3) that we do not want to depend on another external project?

Could this be a performance inhibitor?  I did a small test with 1 KB data, and `murmur2` is x100 slower than `mmh3.hash`

Reply:
No reason. Would you please create a separate issue with this suggestion not to lose it? Let's keep just merge in this PR.